nexus-repository-composer/src/main/java/org/sonatype/nexus/repository/composer/internal/ComposerHostedFacetImpl.java

```diff
@@ -85,8 +85,13 @@ public Content getProviderJson(final String vendor, final String project) throws
   @TransactionalTouchMetadata
   public Content getPackageJson(final String vendor, final String project) throws IOException {
     Content content = content().get(ComposerPathUtils.buildPackagePath(vendor, project));
-    //check composer v1 packages if v2 Path is invalid
-    return content != null ? content : content().get(ComposerPathUtils.buildProviderPath(vendor, project));
+    //Create v2 Package if it´s not existing
+    if (content != null) {
+      return content;
+    } else {
+      rebuildPackageJson(vendor, project);
```

Comment:
Just my two cents: I do not like that we are creating new resources on a GET request.
GET implicates (at least for my understanding and in most of the calls in the repository) a simple: Get me this or say you don't have it.

Now there are `proxy` repositories that cache the fetched result, because that is literally their job.

I do not know if that should also be the case for composer hosted repositories.
But it is the only way I see right now to automatically build a bridge between new and old composer api in a sustainable way.
Without the help of a Sonatype dev, that could point this plugin in the right direction in regards to: rebuild an index during an update of a plugin
*and*
How to properly rebuild the index (and what you should expect from it),

we have no other choice for now, I guess.